### ISSUE TITLE
Vertical Movement Arrows in Snake Game

### ISSUE THREAD

**Issue by MrLatebros** [retrowars/retrowars#15]
I really like the style and feel of the app, so I'll start with a thanks for bringing more retro fun to the FOSS scene.

The only point I might add is that though I thoroughly enjoy playing snake, while the horizontal buttons on the left are intuitive for touchscreen use, to accidentally press 'up' instead of down (or vice-versa), due to the right-side horizontal buttons controlling vertical movement, mars the user experience, especially if it results in an unnecessary game-over.

Perhaps there could be an option to adjust the button positions and/or sizes to suit the user.

I realise that the relevant code is probably this:
[https://github.com/retrowars/retrowars/blob/9c397228fbd1f349e2198ed30b1fc5d08ccb90fe/core/src/com/serwylo/retrowars/games/snake/SnakeGameScreen.kt#L35-L63](https://github.com/retrowars/retrowars/blob/9c397228fbd1f349e2198ed30b1fc5d08ccb90fe/core/src/com/serwylo/retrowars/games/snake/SnakeGameScreen.kt#L35-L63)

And I could alter it and recompile, but it'd be nice to have an option that anyone can use (and still receive updates through F-Droid etc).

If customizable buttons are too much work in code, perhaps just having the vertical movement in snake be with vertical buttons would suffice for me at least.

Thanks for your consideration.

**Comment by pserwylo** [comment]
Thanks for the kind words.

I also actually struggle with the up down buttons! There is no intuitive feel for which one should be up and which down (unlike left and right where my brain is very happy with the one on the left going left and the one on the right going right.

My immediate thought is that I really dislike adding extra options for each feature of the game that could conceivably be altered.

*However*, upon reflection this is actually quite a common thing found in many old console and arcade games. With that in mind it could probably be organised in such a way as to feel right.

How about this:
* [ ] Options menu has sub-screens for each game (but only those which need options)
* [ ] Add an option for controller layout for snake (and also potentially asteroids and Tetris - sometimes I hit drop instead of rotate for tetris).
* [ ] Let the user scroll left or right to previous different controller layouts

Would look something like this:

![layout](https://user-images.githubusercontent.com/248565/121789175-1c893400-cc17-11eb-8ecb-bf1a5a1362a6.png)

Thoughts?

**Comment by pserwylo** [comment]
Potential layouts for Tetris.

Perhaps a few too many, but I guess once the code to customize this is in place, it is not hard to add others, and once you choose one you like you will never return to this screen again, so it doesn't really degrade the user experience to provide more options...

**Layout A**
```
+-----------------------------+
|                             |
|  [L] [R]           [U] [D]  |
+-----------------------------+
```

**Layout B**
```
+-----------------------------+
|                             |
|  [L] [R]           [D] [U]  |
+-----------------------------+
```

**Layout C**
```
+-----------------------------+
|                             |
|  [L] [U]           [D] [R]  |
+-----------------------------+
```

**Layout D**
```
+-----------------------------+
|                             |
|  [L] [D]           [U] [R]  |
+-----------------------------+

```

**Layout E**
```
+-----------------------------+
|      [U]                    |
|  [L] [D] [R]                |
+-----------------------------+
```

**Layout F**
```
+-----------------------------+
|                    [U]      |
|                [L] [D] [R]  |
+-----------------------------+
```

**Layout G**
```
+-----------------------------+
|             [U]             |
|         [L] [D] [R]         |
+-----------------------------+
```

**Comment by MrLatebros** [comment]
Thanks for your considered response. Your idea of a simple list of common layout options would work well I think.

I'm conscious of balancing customisation and features with code simplicity and coding effort, especially as it isn't **my** time and effort, so however you might see fit to implement it, I'd agree that simpler is better.

> once the code to customize this is in place, it is not hard to add others, and once you choose one you like you will never return to this screen again

Yeah, exactly.

Thanks again for such a considered response in such a prompt reply.

**Comment by pserwylo** [comment]
Alrighty, here we go!

https://user-images.githubusercontent.com/248565/121824387-457cf800-ccef-11eb-8eeb-f232fed1539b.mp4

[Relatively straightforward to add new layouts too (and will allow for custom ones in the future as I just hand-rolled a little text-based format)](https://github.com/retrowars/retrowars/blob/352eebadf298b1f3e4d656f74246448568fce828/core/src/com/serwylo/retrowars/input/SoftController.kt#L129-L164):

Example:

```
[<---------->][      ][  up  ][       ]
[<---------->][ left ][ down ][ right ]
```

v0.3.0 is currently being built by F-Droid ready for release soon. This change is in v0.4.0, which will be released after that, so perhaps in a week or two.

Thanks again for your input @MrLatebros. When you see 0.4.0 drop with this on F-Droid, it would be great to hear back from you as to whether his is what you had in mind, and whether it solved the confusion both you and I were feeling when playing snake.
